command/agent/agent.go

```diff
@@ -343,6 +343,13 @@ func convertServerConfig(agentConfig *Config) (*nomad.Config, error) {
 	conf.JobMaxPriority = jobMaxPriority
 	conf.JobDefaultPriority = jobDefaultPriority
 
+	if agentConfig.Server.JobTrackedVersions != nil {
+		if *agentConfig.Server.JobTrackedVersions < 0 {
+			return nil, fmt.Errorf("invalid Config, job_tracked_versions must be non-negative")
```

Comment:
The caller wraps this more nicely, so let's be consistent with the nearby message formats:

```suggestion
			return nil, fmt.Errorf("job_tracked_versions must be non-negative")
```

But also, is setting the value to 0 actually safe? There aren't any tests added that would exercise that case, which seems important to make sure we get right.

Reply:
@tgross setting to 0 is not safe, I have updated the condition to update this.